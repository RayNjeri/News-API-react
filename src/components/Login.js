import React from 'react';
import { GoogleLogin } from 'react-google-login-component';



export default class Login extends React.Component {

    constructor(props, context) {
        super(props, context);
    }

    responseGoogle(googleUser) {
        var id_token = googleUser.getAuthResponse().id_token;
        console.log({ accessToken: id_token });

    }

    render() {
        return (
            <div>
                <GoogleLogin socialId="300991801998-8s16am8jt76ou8pge5iisdih6dokqss5"
                    class="google-login"
                    scope="profile"
                    responseHandler={this.responseGoogle}
                    buttonText="Login With Google" />
            </div>
        );
    }

}


