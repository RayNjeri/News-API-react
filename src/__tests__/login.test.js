import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Login from '../components/Login';

describe('Login Component', () => {
    it('Snapshot of Login Component', () => {
        const login = shallow(<Login />);
        const tree = shallowToJSON(login);
        expect(tree).toMatchSnapshot();
    });
    // it('should displays log in button', () => {
    //     const renderedComponent = shallow(<Button></Button>);
    //     expect(renderedComponent.find("button").node).toBeDefined();
    // });
    it('should display log in text in the button', () => {
        const text = 'Log in with google';
        const renderedComponent = shallow(<Button>{'LOG IN'}</Button>);
        expect(renderedComponent.contains(text)).toEqual(true);
    });
});

