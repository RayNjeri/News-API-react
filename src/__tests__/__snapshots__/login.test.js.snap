// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Snapshot of Login Component 1`] = `
<div>
  <GoogleLogin
    buttonText="Login With Google"
    class="google-login"
    responseHandler={[Function]}
    scope="profile"
    socialId="300991801998-8s16am8jt76ou8pge5iisdih6dokqss5"
  />
</div>
`;
